public/app/core/logsModel.ts

```diff
@@ -459,7 +459,8 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
         entry,
         raw: message,
         labels: labels || {},
-        uid: idField ? idField.values[j] : j.toString(),
+        // Append refId to uid to make it unique across all series in a case when series contain duplicates
+        uid: `${series.refId}_${idField ? idField.values[j] : j.toString()}`,
```

Comment:
I am curious if we should append the `refId` in order to generate same IDs at least for Loki with that change?

I don't have a strong preference, just wanted to raise that.

Reply:
You are right!

Reply:
Fixed in https://github.com/grafana/grafana/pull/68569/commits/f2991c4b0a633c1d8446bbfd8e19fc3970a21ec6. I like `refId` being the first, so I updated lokiLiveStream code